nixos/modules/services/hardware/logiops.nix

```diff
@@ -0,0 +1,139 @@
+                                mode: "OnRelease";
+                                action =
+                                {
+                                    type: "Keypress";
+                                    keys: ["KEY_UP"];
+                                };
+                            },
+                            {
+                                direction: "Down";
+                                mode: "OnRelease";
+                                action =
+                                {
+                                    type: "Keypress";
+                                    keys: ["KEY_DOWN"];
+                                };
+                            },
+                            {
+                                direction: "Left";
+                                mode: "OnRelease";
+                                action =
+                                {
+                                    type: "CycleDPI";
+                                    dpis: [400, 600, 800, 1000, 1200, 1400, 1600];
+                                };
+                            },
+                            {
+                                direction: "Right";
+                                mode: "OnRelease";
+                                action =
+                                {
+                                    type = "ToggleSmartshift";
+                                }
+                            },
+                            {
+                                direction: "None"
+                                mode: "NoPress"
+                            }
+                        );
+                    };
+                },
+                {
+                    cid: 0xc4;
+                    action =
+                    {
+                        type: "Keypress";
+                        keys: ["KEY_A"];
+                    };
+                }
+            );
+        }
+        );
+      '';
+    };
+
+    # settings = mkOption {
+    #   type = types.submodule {
+    #     # Declare that the settings option supports arbitrary format values, libconfig here
+    #     freeformType = settingsFormat.type;
+    #   };
+    #   default = {};
+    #   # Add upstream documentation to the settings description
+    #   description = ''
+    #     Configuration for `logiops`, see
+    #     <link xlink:href="https://github.com/PixlOne/logiops/wiki/Configuration"/>
+    #     for supported values.
+    #   '';
+    # };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.logiops = {
+      description = "Logitech Configuration Daemon";
+      startLimitIntervalSec = 0;
+      serviceConfig = {
+        Type = "simple";
+        ExecStart = "${pkgs.logiops}/bin/logid";
+        User = "root";
+        ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
+        Restart = "on-failure";
+      };
+      wantedBy = [ "multi-user.target" ];
+      restartTriggers = [ pkgs.logiops ];
```

Comment:
This is redundant. Maybe you wanted `cfg.extraConfig` here, but I'm not sure.

Reply:
Is it redundant because I'm already referencing the package in the `ExecStart` line?

Restarting on `cfg.extraConfig` is a good idea, as a user I would expect changes to the config to take effect once I've rebuilt NixOS (currently it doesn't).

Reply:
Exactly! You understand perfectly now 👍